data/src/main/java/com/linkedin/data/codec/symbol/DefaultSymbolTableProvider.java

```diff
@@ -48,6 +53,16 @@
    */
   private static final String ACCEPT_HEADER = "Accept";
 
+  /**
+   * IC header key
+   */
+  private static final String IC_HEADER = "X-LI-R2-W-IC-1";
+
+  /**
+   * Default IC header
+   */
+  private static final String DEFAULT_IC_HEADER = "serviceCallTraceData=%7B%22caller%22%3A%7B%22container%22%3A%22container%22%2C%22service%22%3A%22restli%22%2C%22env%22%3A%22env%22%2C%22instance%22%3A%22inst%22%2C%22machine%22%3A%22machine.linkedin.biz%22%2C%22version%22%3A%220.0.1%22%7D%2C%22treeId%22%3A%22{}%22%2C%22context%22%3A%7B%22source%22%3A%22restli%22%2C%22forceTraceEnabled%22%3A%22false%22%2C%22debugEnabled%22%3A%22false%22%2C%22traceGroupingKey%22%3A%22restli-default-symboltable%22%7D%7D,com.linkedin.container.rpc.trace.rpcTrace=(machine.linkedin.biz,restli,1970/01/01 01:00:00.000)";
```

Comment:
can this be more generic? For example, to be Injected by some customHeaderInjector in Container? this is a open source repo and we should bring minimal concept of IC

Reply:
couldn't be injected at boot time because dynamic tree id required in trace for each request.

Reply:
Injected by a factory class in Container doesn't have to be in boot time, right? It can be in each request.

I am just thinking about trying not to add linkedin specific thing here

Reply:
WDYT @zhang-chris

Reply:
updated